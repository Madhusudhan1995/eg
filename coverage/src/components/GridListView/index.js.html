<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/GridListView/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/GridListView</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/82</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/32</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/65</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {connect} from "react-redux";<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span></span></span></span></span></span></span></span>
import React,<span class="cstat-no" title="statement not covered" > {Component} from "react";</span>
import {ScrollView,<span class="cstat-no" title="statement not covered" > View, Text, Image, Butt</span>on, Dimensions, LayoutAnimation, UIManager, TouchableOpacity} from "react-native";
&nbsp;
import LinkButton <span class="cstat-no" title="statement not covered" >from "../LinkButton";</span>
import {dpToPx}<span class="cstat-no" title="statement not covered" > from "../../helpers";</span>
&nbsp;
import {styles<span class="cstat-no" title="statement not covered" >, renderStyles} from</span> "./styles";<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
const defaultProps <span class="cstat-no" title="statement not covered" >= {</span>
    voucherList: []
}
&nbsp;
export default <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >cl</span>ass GridListView exte<span class="cstat-no" title="statement not covered" >nds Component {</span></span>
&nbsp;
    constru<span class="fstat-no" title="function not covered" >ctor(props) {<span class="cstat-no" title="statement not covered" ></span></span>
<span class="cstat-no" title="statement not covered" >        super(props);<span class="cstat-no" title="statement not covered" ></span></span>
        this.state = {
            viewStyle: "gridView",
            imageHeight: 120
        }
    }
&nbsp;
    onListView = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >        this.setState({</span>
            viewStyle: "listView"
<span class="fstat-no" title="function not covered" ></span>        }, () =&gt; {
<span class="cstat-no" title="statement not covered" >            this._animate(</span>);
        });
    }<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
    onGridView = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >      this.setState({</span>
          viewStyle: "gridView"
<span class="fstat-no" title="function not covered" >      },</span> () =&gt; {
<span class="cstat-no" title="statement not covered" >          this._animate();</span>
      });
    }<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
    measureView = <span class="fstat-no" title="function not covered" >ev</span>ent =&gt; {
<span class="cstat-no" title="statement not covered" >        this.setState({</span>
            imageHeight: event.nativeEvent.layout.height
        });
    }<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
    _animate = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >        UIManager.setLayoutAnimationEnabledExperimental &amp;&amp; UIManager.setLayoutAnimationEnabledExperimental(true);</span>
<span class="cstat-no" title="statement not covered" >        LayoutAnimation.easeInEaseOut();</span>
    }<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
    arrayChunking = <span class="fstat-no" title="function not covered" >(n</span>um, arr) =&gt; {
        const newAr<span class="cstat-no" title="statement not covered" >r =</span> [];
        let count <span class="cstat-no" title="statement not covered" >= </span>0;
<span class="cstat-no" title="statement not covered" >        arr.forEac<span class="fstat-no" title="function not covered" >h(</span>el =&gt; {</span>
<span class="cstat-no" title="statement not covered" >            if(newArr.length &gt; 0 &amp;&amp; count % num !== 0) {</span>
<span class="cstat-no" title="statement not covered" >                newArr[newArr.length - 1].push</span>(el);
            } else {
<span class="cstat-no" title="statement not covered" >                newArr.push([</span>el]);
            }
<span class="cstat-no" title="statement not covered" >            count</span>++;
        });
<span class="cstat-no" title="statement not covered" >        return newArr</span>;
    }<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
    renderListContainer = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
        const {vie<span class="cstat-no" title="statement not covered" >wStyle} = this.state;</span>
<span class="cstat-no" title="statement not covered" >        return this.arrayChunking(2, this.props.voucherList).map(<span class="fstat-no" title="function not covered" >(i</span>tem, index) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >            return(&lt;View key={index} style={renderStyles(viewStyle).gridContainer}&gt;</span>
                {this.renderListItem(item)}
            &lt;/View&gt;);
        });
    }<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
    renderListItem = <span class="fstat-no" title="function not covered" >(i</span>tems) =&gt; {
        const imageContWidt<span class="cstat-no" title="statement not covered" >h = Dimensions.get('window').width/2 - dpToPx(7);</span>
<span class="cstat-no" title="statement not covered" >        const {vie</span>wStyle, imag<span class="cstat-no" title="statement not covered" >eHeight} = this.state;<span class="cstat-no" title="statement not covered" ></span></span>
<span class="cstat-no" title="statement not covered" >        return items.ma<span class="fstat-no" title="function not covered" >p(</span>(item, index) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >            return(</span>
                &lt;TouchableOpacity key={index} style={renderStyles(viewStyle).gridItemContainer} onPress={() =&gt; this.onCardClick(this.state.outletImg)}&gt;<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span>
            &lt;View style={[{transform: [{translateX: viewStyle === "gridView" ? imageContWidth : 0}]}, renderStyles(viewStyle).gridImageContainer]}&gt;
                &lt;Image source={{uri : item.image}} style={renderStyles(viewStyle, imageHeight).gridImageStyle} resizeMode="cover"/&gt;
            &lt;/View&gt;
            &lt;View onLayout={(event) =&gt; this.measureView(event)}  style={[{trans<span class="fstat-no" title="function not covered" >form: [{t</span>ranslate<span class="cstat-no" title="statement not covered" >X: viewStyle === "gridView" ? -im</span>ageContWidth : 0}]}, renderStyles(viewStyle).gridItemContent]}&gt;
                &lt;Text style={styles.vouchersStoreName}&gt;{item.vendor}&lt;/Text&gt;
                &lt;Text style={styles.vouchersOfferPrice}&gt;{item.code}&lt;/Text&gt;
                &lt;Text style={styles.vouchersOfferText}&gt;{item.desc}&lt;/Text&gt;
                &lt;Text style={styles.vouchersValidText}&gt;{this.renderDateFromSecs(item.validityends)}&lt;/Text&gt;
            &lt;/View&gt;
        &lt;/TouchableOpacity&gt;
        )
        })
    }<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
    renderDateFromSecs = <span class="fstat-no" title="function not covered" >(m</span>illisecs) =&gt; {
<span class="cstat-no" title="statement not covered" >        if(millisecs === 0 || millisecs === null) {</span>
<span class="cstat-no" title="statement not covered" >            return </span>"";
        }
        var date <span class="cstat-no" title="statement not covered" >= new Date(millisecs</span>);
        var textDate <span class="cstat-no" title="statement not covered" >= date.toString("DD MMM YYYY"</span>);
<span class="cstat-no" title="statement not covered" >        return "Valid till " + date.getDay() + " " + this.renderMonthString(date.getMonth()) + " " + date.getFullYear();</span>
    }<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
    renderMonthString = <span class="fstat-no" title="function not covered" >(m</span>onthNum) =&gt; {
<span class="cstat-no" title="statement not covered" >        switch(monthNum) {</span>
            case 0:
<span class="cstat-no" title="statement not covered" >            return "JAN"</span>;
            case 1:
<span class="cstat-no" title="statement not covered" >            return "FEB"</span>;
            case 2:
<span class="cstat-no" title="statement not covered" >            return "MAR"</span>;
            case 3:
<span class="cstat-no" title="statement not covered" >            return "APR"</span>;
            case 4:
<span class="cstat-no" title="statement not covered" >            return "MAY"</span>;
            case 5:
<span class="cstat-no" title="statement not covered" >            return "JUN"</span>;
            case 6:
<span class="cstat-no" title="statement not covered" >            return "JUL"</span>;
            case 7:
<span class="cstat-no" title="statement not covered" >            return "AUG"</span>;
            case 8:
<span class="cstat-no" title="statement not covered" >            return "SEP"</span>;
            case 9:
<span class="cstat-no" title="statement not covered" >            return "OCT"</span>;
            case 10:
<span class="cstat-no" title="statement not covered" >            return "NOV"</span>;
            case 11:
<span class="cstat-no" title="statement not covered" >            return "DEC"</span>;
        }
    }<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span>
&nbsp;
    render() {
        const {vie<span class="cstat-no" title="statement not covered" >wStyle} = this.state;</span>
<span class="cstat-no" title="statement not covered" >        return(</span>
          &lt;View style={renderStyles(viewStyle).container}&gt;
              &lt;View style={renderStyles(viewStyle).buttonContainer}&gt;
                  &lt;LinkButton
                      title="List View"
                      onPress={this.onGridView}
                      color={viewStyle === "listView" ? "#999999" : "#000000"}
                      containerStyle={renderStyles(viewStyle).listViewButton}
                      iconPath={require("../../assets/images/listViewIcon.png")}
                      iconStyle={{width: dpToPx(7), height: "auto", marginRight: dpToPx(4)}} /&gt;
                  &lt;LinkButton
                      title="Grid View"
                      color={viewStyle === "gridView" ? "#999999" : "#000000"}
                      onPress={this.onListView}
                      containerStyle={renderStyles(viewStyle).gridViewButton}
                      iconPath={require("../../assets/images/gridViewIcon.png")}
                      iconStyle={{width: dpToPx(7), height: "auto", marginRight: dpToPx(4)}} /&gt;
              &lt;/View&gt;
              &lt;ScrollView&gt;
                  &lt;View style={renderStyles(viewStyle).gridContContainer}&gt;
                      {this.props.voucherList.length &gt; 0 &amp;&amp; this.renderListContainer()}
                  &lt;/View&gt;
              &lt;/ScrollView&gt;
          &lt;/View&gt;
        );
    }<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span>
}
&nbsp;
<span class="cstat-no" title="statement not covered" >GridListView.defaultProps = defaultProps;</span>
&nbsp;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Dec 05 2018 19:09:36 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
