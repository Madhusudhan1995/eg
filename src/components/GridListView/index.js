import {connect} from "react-redux";
import React, {Component} from "react";
import {ScrollView, View, Text, Image, Button, Dimensions, LayoutAnimation, UIManager, TouchableOpacity} from "react-native";

import LinkButton from "../LinkButton";
import {dpToPx} from "../../helpers";

import {styles, renderStyles} from "./styles";

const defaultProps = {
    voucherList: []
}

export default class GridListView extends Component {

    constructor(props) {
        super(props);
        this.state = {
            viewStyle: "gridView",
            imageHeight: 120
        }
    }

    onListView = () => {
        this.setState({
            viewStyle: "listView"
        }, () => {
            this._animate();
        });
    }

    onGridView = () => {
      this.setState({
          viewStyle: "gridView"
      }, () => {
          this._animate();
      });
    }

    measureView = event => {
        this.setState({
            imageHeight: event.nativeEvent.layout.height
        });
    }

    _animate = () => {
        UIManager.setLayoutAnimationEnabledExperimental && UIManager.setLayoutAnimationEnabledExperimental(true);
        LayoutAnimation.easeInEaseOut();
    }

    arrayChunking = (num, arr) => {
        const newArr = [];
        let count = 0;
        arr.forEach(el => {
            if(newArr.length > 0 && count % num !== 0) {
                newArr[newArr.length - 1].push(el);
            } else {
                newArr.push([el]);
            }
            count++;
        });
        return newArr;
    }

    renderListContainer = () => {
        const {viewStyle} = this.state;
        return this.arrayChunking(2, this.props.voucherList).map((item, index) => {
            return(<View key={index} style={renderStyles(viewStyle).gridContainer}>
                {this.renderListItem(item)}
            </View>);
        });
    }

    renderListItem = (items) => {
        const imageContWidth = Dimensions.get('window').width/2 - dpToPx(7);
        const {viewStyle, imageHeight} = this.state;
        return items.map((item, index) => {
            return(
                <TouchableOpacity key={index} style={renderStyles(viewStyle).gridItemContainer} onPress={() => this.onCardClick(this.state.outletImg)}>
            <View style={[{transform: [{translateX: viewStyle === "gridView" ? imageContWidth : 0}]}, renderStyles(viewStyle).gridImageContainer]}>
                <Image source={{uri : item.image}} style={renderStyles(viewStyle, imageHeight).gridImageStyle} resizeMode="cover"/>
            </View>
            <View onLayout={(event) => this.measureView(event)}  style={[{transform: [{translateX: viewStyle === "gridView" ? -imageContWidth : 0}]}, renderStyles(viewStyle).gridItemContent]}>
                <Text style={styles.vouchersStoreName}>{item.vendor}</Text>
                <Text style={styles.vouchersOfferPrice}>{item.code}</Text>
                <Text style={styles.vouchersOfferText}>{item.desc}</Text>
                <Text style={styles.vouchersValidText}>{this.renderDateFromSecs(item.validityends)}</Text>
            </View>
        </TouchableOpacity>
        )
        })
    }

    renderDateFromSecs = (millisecs) => {
        if(millisecs === 0 || millisecs === null) {
            return "";
        }
        var date = new Date(millisecs);
        var textDate = date.toString("DD MMM YYYY");
        return "Valid till " + date.getDay() + " " + this.renderMonthString(date.getMonth()) + " " + date.getFullYear();
    }

    renderMonthString = (monthNum) => {
        switch(monthNum) {
            case 0:
            return "JAN";
            case 1:
            return "FEB";
            case 2:
            return "MAR";
            case 3:
            return "APR";
            case 4:
            return "MAY";
            case 5:
            return "JUN";
            case 6:
            return "JUL";
            case 7:
            return "AUG";
            case 8:
            return "SEP";
            case 9:
            return "OCT";
            case 10:
            return "NOV";
            case 11:
            return "DEC";
        }
    }

    render() {
        const {viewStyle} = this.state;
        return(
          <View style={renderStyles(viewStyle).container}>
              <View style={renderStyles(viewStyle).buttonContainer}>
                  <LinkButton
                      title="List View"
                      onPress={this.onGridView}
                      color={viewStyle === "listView" ? "#999999" : "#000000"}
                      containerStyle={renderStyles(viewStyle).listViewButton}
                      iconPath={require("../../assets/images/listViewIcon.png")}
                      iconStyle={{width: dpToPx(7), height: "auto", marginRight: dpToPx(4)}} />
                  <LinkButton
                      title="Grid View"
                      color={viewStyle === "gridView" ? "#999999" : "#000000"}
                      onPress={this.onListView}
                      containerStyle={renderStyles(viewStyle).gridViewButton}
                      iconPath={require("../../assets/images/gridViewIcon.png")}
                      iconStyle={{width: dpToPx(7), height: "auto", marginRight: dpToPx(4)}} />
              </View>
              <ScrollView>
                  <View style={renderStyles(viewStyle).gridContContainer}>
                      {this.props.voucherList.length > 0 && this.renderListContainer()}
                  </View>
              </ScrollView>
          </View>
        );
    }
}

GridListView.defaultProps = defaultProps;

