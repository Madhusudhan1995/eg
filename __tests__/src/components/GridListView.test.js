import React from "react";
import {View, TouchableOpacity} from "react-native";
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import GridListView from "../../../src/components/GridListView/index";

describe("GridListView Component", () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<GridListView />)
    })
    it("should render component", () => {
        expect(wrapper.find(View)).toHaveLength(3);
    });

    it("should call TouchableOpacity", () => {
        expect(wrapper.find(TouchableOpacity)).toHaveLength(0);
    });
    it('ListPicker should render correctly', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot()
    });

    // it("should call onGridView ", () => {
    //     const onGridView = jest.fn();
    //     GridListView.prototype.onGridView = onGridView;
    //     const wrapper = shallow(<GridListView />);
    //     wrapper.instance().onGridView ();
    //     expect(wrapper.length).toEqual(1);
    // });
    
    // it("should call _animate", () => {
    //     const _animate = jest.fn();
    //     GridListView.prototype._animate = _animate;
    //     const wrapper = shallow(<GridListView />);
    //     wrapper.instance()._animate();
    //     expect(wrapper.length).toEqual(1);
    // });
})
