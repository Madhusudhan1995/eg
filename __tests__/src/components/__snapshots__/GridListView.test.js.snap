// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridListView Component ListPicker should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderBottomColor": "#dddddd",
        "borderBottomWidth": 1,
        "borderTopColor": "#dddddd",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "paddingVertical": 8,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRightColor": "#dddddd",
          "borderRightWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/images/listViewIcon.png",
          }
        }
        style={
          Object {
            "height": "auto",
            "marginRight": 8,
            "width": 14,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 16,
            },
            Object {
              "color": "#000000",
            },
            Object {},
          ]
        }
      >
        List View
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/images/gridViewIcon.png",
          }
        }
        style={
          Object {
            "height": "auto",
            "marginRight": 8,
            "width": 14,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 16,
            },
            Object {
              "color": "#999999",
            },
            Object {},
          ]
        }
      >
        Grid View
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "paddingTop": 12,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
